### ISSUE TITLE
Eventloop leak warnings from modules

### ISSUE THREAD

**Issue by richteer** [Halibot/halibot#66]
Example of warning from Travis:

```
../opt/python/3.5.2/lib/python3.5/asyncio/base_events.py:429: ResourceWarning: unclosed event loop <_UnixSelectorEventLoop running=False closed=False debug=False>

  warnings.warn("unclosed event loop %r" % self, ResourceWarning)
```

It appears the issue is in `HalObject`'s `_shutdown()`, here is an example patch to mitigate the leak warnings.

``` diff
diff --git a/halibot/halobject.py b/halibot/halobject.py
index d1007a7..d136b8a 100644
--- a/halibot/halobject.py
+++ b/halibot/halobject.py
@@ -18,6 +18,7 @@ class HalObject():
        def _shutdown(self):
                self.eventloop.call_soon_threadsafe(self.eventloop.stop)
                self._thread.join()
+               self.eventloop.close()
                self.shutdown()

        def _queue_msg(self, msg):
```

_THAT SAID_, I marked this as RFC, because this is partially undefined behavior. If for example, a module is `shutdown` then re-`init`'d, should it create a new event loop? In this case, we can call `_shutdown()`, which closes the loop, but `init()` will _NOT_ recreate a new one (only `__init__()` does this.

I see two options:
- `eventloop` remains being created in `.__init__()`, started in `.init()`, stopped in `._shutdown()`, and only `.close()`'d in `Halibot.shutdown()`. This means that a module can continue to receive messages to the event loop, but will not process them (another potential soft-leak problem).
- The `eventloop` is created when `.init()` is called, and `.close()`'d in `._shutdown()`. This does mean that there will probably be an exception if that module continues to receive messages, but this at least ensures when the module is stopped by any means, the eventloop is cleared completely.

**Comment by sjrct** [comment]
I am fond of the idea of something cleaning up what it created, and like this fix as it is. We could, perhaps, implement a `restart` method of some kind that stops the event loop without closing it, does restart logic, and then opens it back up.

**Comment by richteer** [comment]
+1 to the restart method. I think we should assume that if a module is stopped, it shouldn't be started again. The exception to this case is reloading, however we can treat that as a full tear-down and bring-up, thus using a new eventloop.

**Comment by richteer** [comment]
Fixed with #81
